kasa/transports/linkietransport.py

```diff
@@ -0,0 +1,137 @@
+"""Implementation of the legacy TP-Link Smart Home Protocol.
+
+Encryption/Decryption methods based on the works of
+Lubomir Stroetmann and Tobias Esser
+
+https://www.softscheck.com/en/reverse-engineering-tp-link-hs110/
+https://github.com/softScheck/tplink-smartplug/
+
+which are licensed under the Apache License, Version 2.0
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+from __future__ import annotations
+
+import logging
+import base64
+import ssl
+from urllib.parse import quote
+from yarl import URL
+
+from kasa.httpclient import HttpClient
+from kasa.deviceconfig import DeviceConfig
+from kasa.exceptions import KasaException, _RetryableError
+from kasa.json import loads as json_loads
+from kasa.credentials import DEFAULT_CREDENTIALS, Credentials, get_default_credentials
+from kasa.transports.xortransport import XorEncryption
+
+from .basetransport import BaseTransport
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LinkieTransport(BaseTransport):
```

Comment:
If I understand correctly this is slightly different from the implementation in https://github.com/python-kasa/python-kasa/pull/537 as it uses the `LINKIE2.json` which passes a credentials_hash in the header instead of using basic auth? So perhaps this class should be called `LinkieTransportV2` like we do for klap?

Do you have a discovery response for your device you could add to the PR comments?

Reply:
V2 also uses basic auth, but I directly add the header + hash, while the [older implementation](https://github.com/bstrdsmkr/python-kasa/blob/kasacam_support/kasa/smartcameraprotocol.py#L40) used the `httpx` client to handle (I assume it adds the same header).

[It seems the default credentials](https://github.com/python-kasa/python-kasa/pull/537/files#r1444138443) for V1 are `admin:admin`, though I can't confirm myself. In addition from what I'm seeing, V1 uses HTTP while V2 uses HTTPS? Those may be the only differences, I'm not sure if that warrants 2 separate classes (with additional logic for switching between transports).

Reply:
I think it should be fine to have a base class that V2 derives from and overrides methods as necessary. I don't think that's needed for this PR though but we should still name this class `LinkieTransportV2` as we know it's V2 and we could put V1 in later if necessary.

Were you able to get hold of a discovery response? Is it exactly the same as the `system` section in the fixture or is there anything else there that could be used to indicate which version it's using? A lot of users prevent their devices from updating firmware, so we should hopefully be able to find a way to say that the older transport is not supported.

Reply:
Updated the class name to include `V2`. I also checked the discover result and yea, it's identical to the fixture.

Looking at the Android app, `stream_version` is used to determine the Linkie endpoint. Do we just need a version check where `LinkieTransportV2` is being assigned inside `DeviceConfig`?